src/foam/dao/ArrayDAO.js

```diff
@@ -32,8 +32,8 @@ foam.CLASS({
       class: 'Class',
       name: 'of',
       factory: function() {
-        if (this.array.length === 0) return this.lookup('foam.core.FObject');
-        return this.array[0].cls_;
+        if ( this.array.length === 0 ) return this.lookup('foam.core.FObject');
+        return undefined;
```

Comment:
Factories aren't supposed to return undefined.

Reply:
Is there a safe alternative in this case? Should I use `null`?